classes/class-wc-connect-compatibility.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * A class for working around the quircks and different versions of WordPress/WooCommerce
+ */
+
+// No direct access please
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'WC_Connect_Compatibility' ) ) {
+
+	class WC_Connect_Compatibility {
+		private static $singleton;
+
+		public static function instance() {
+			if( self::$singleton === null ) {
+				self::$singleton = new WC_Connect_Compatibility();
+			}
+
+			return self::$singleton;
+		}
+
+		public function __construct() {
+		}
```

Comment:
The singleton is more future-proof compared to a static class. Right now the class is fairly simple, and either one would work, but in the future I can see this class getting bigger, perhaps needing a constructor to initialize values, or even an interface to load different versions of a base class depending on the version of WooCommerce being used.